web/server/utils.py

```diff
@@ -26,15 +26,42 @@ def __init__(self, *args: t.Any, **kwargs: t.Any) -> None:
         super().__init__(*args, **kwargs)
 
 
-async def run_in_executor(func: t.Callable[..., R], *args: t.Any) -> R:
+async def run_in_executor(
+    func: t.Callable[..., R], *args: t.Any, exception: t.Optional[ApiException] = None
+) -> R:
     """Run in the default loop's executor"""
 
     @functools.wraps(func)
     def func_wrapper() -> R:
         try:
             return func(*args)
+        except ApiException as e:
+            api_console.log_exception(e)
+            raise e
         except Exception as e:
-            api_console.log_exception()
+            message = "An unexpected error occurred"
+            origin = "API -> utils -> run_in_executor"
+            trigger = "An unexpected error occurred"
+            status_code = HTTP_422_UNPROCESSABLE_ENTITY
+
+            if exception:
+                if exception.message:
+                    message = exception.message
+                if exception.origin:
+                    origin = exception.origin
+                if exception.trigger:
+                    trigger = exception.trigger
+                if exception.status_code:
+                    status_code = exception.status_code
+
+            api_console.log_exception(
+                ApiException(
```

Comment:
we need to re-instantiate `ApiException` here to catch actual traceback of an error, because that's where error actually occurred